src/cascadia/TerminalCore/TerminalSelection.cpp

```diff
@@ -235,92 +73,173 @@ const bool Terminal::IsCopyOnSelectActive() const noexcept
+    // set the selection pivot to expand the selection using SetSelectionEnd()
+    _selectionPivot = _ConvertToBufferCell(viewportPos);
+    _buffer->GetSize().Clamp(_selectionPivot);
 
-    // scan rightwards until delimiter is found and
-    // set endSelectionPosition to one left of that spot
-    _endSelectionPosition = _ExpandDoubleClickSelectionRight(positionWithOffsets);
-    THROW_IF_FAILED(ShortSub(_endSelectionPosition.Y, gsl::narrow<SHORT>(ViewStartIndex()), &_endSelectionPosition.Y));
+    _multiClickSelectionMode = expansionMode;
+    SetSelectionEnd(viewportPos);
 
-    _selectionActive = true;
-    _multiClickSelectionMode = SelectionExpansionMode::Word;
+    // we need to set the _selectionPivot again
+    // for future shift+clicks
+    _selectionPivot = _selectionStart;
 }
 
 // Method Description:
-// - Select the entire row of the position clicked
+// - Record the position of the beginning of a selection
 // Arguments:
 // - position: the (x,y) coordinate on the visible viewport
-void Terminal::TripleClickSelection(const COORD position)
+void Terminal::SetSelectionAnchor(const COORD viewportPos)
 {
-    SetSelectionAnchor({ 0, position.Y });
-    SetEndSelectionPosition({ _buffer->GetSize().RightInclusive(), position.Y });
+    _selectionPivot = _ConvertToBufferCell(viewportPos);
+    _buffer->GetSize().Clamp(_selectionPivot);
+
+    _multiClickSelectionMode = SelectionExpansionMode::Cell;
 
-    _multiClickSelectionMode = SelectionExpansionMode::Line;
+    // Unlike MultiClickSelection(), we need to set
+    // these vars BEFORE SetSelectionEnd()
+    _allowSingleCharSelection = (_copyOnSelect) ? false : true;
+    _selectionStart = _selectionPivot;
+
+    SetSelectionEnd(viewportPos);
 }
 
 // Method Description:
-// - Record the position of the beginning of a selection
+// - Update selection anchors when dragging to a position
+// - based on the selection expansion mode
 // Arguments:
-// - position: the (x,y) coordinate on the visible viewport
-void Terminal::SetSelectionAnchor(const COORD position)
+// - viewportPos: the (x,y) coordinate on the visible viewport
+// - expansionModeOverwrite: overwrites the _multiClickSelectionMode for this function call. Used for ShiftClick
+void Terminal::SetSelectionEnd(const COORD viewportPos, std::optional<SelectionExpansionMode> expansionModeOverwrite)
 {
-    _selectionAnchor = position;
+    auto textBufferPos = _ConvertToBufferCell(viewportPos);
+    _buffer->GetSize().Clamp(textBufferPos);
 
-    // include _scrollOffset here to ensure this maps to the right spot of the original viewport
-    THROW_IF_FAILED(ShortSub(_selectionAnchor.Y, gsl::narrow<SHORT>(_scrollOffset), &_selectionAnchor.Y));
+    // if this is a shiftClick action, we need to overwrite the _multiClickSelectionMode value (even if it's the same)
+    // Otherwise, we may accidentally expand during other selection-based actions
+    const auto shiftClick = expansionModeOverwrite.has_value();

```

Comment:
I don't love how the presence of this mcdude is used to determine _whether_ the user shift-clicked. is there a better name than `shiftClick` for this?